python/paddle/fleet/meta_optimizers/amp_optimizer.py

```diff
@@ -0,0 +1,62 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+
+from paddle.fluid.contrib.mixed_precision.decorator import decorate
+from paddle.fluid.contrib.mixed_precision.fp16_lists import AutoMixedPrecisionLists
+from .meta_optimizer_base import MetaOptimizerBase
+
+__all__ = ["AMPOptimizer"]
+
+
+class AMPOptimizer(MetaOptimizerBase):
+    def __init__(self, optimizer):
+        super(AMPOptimizer, self).__init__(optimizer)
+        self.inner_opt = optimizer
+        self.amp_opt = None
+        # we do not allow meta optimizer to be inner optimizer currently
+        self.meta_optimizers_white_list = []
+
+    def _set_basic_info(self, loss, role_maker, user_defined_optimizer,
+                        user_defined_strategy):
+        super(AMPOptimizer, self)._set_basic_info(
+            loss, role_maker, user_defined_optimizer, user_defined_strategy)
+
+    def _can_apply(self):
+        if self.user_defined_strategy.amp:
+            return True
+        return False
+
+    def minimize_impl(self,
+                      loss,
+                      startup_program=None,
+                      parameter_list=None,
+                      no_grad_set=None):
+        if self.amp_opt is None:
+            config = self.user_defined_strategy.amp_configs
+            print(config)
+            custom_white_list = set(config['custom_white_list'])
+            custom_black_list = set(config['custom_black_list'])
+            custom_black_varnames = set(config['custom_black_varnames'])
+            amp_lists = AutoMixedPrecisionLists(
+                custom_white_list, custom_black_list, custom_black_varnames)
+
+            self.amp_opt = decorate(
```

Comment:
`decorate` is too common to directly import, which may pollute the namespace in my opinion. It is better to use the full path `paddle.fluid.contrib.mixed_precision.decorator` or `mixed_precision.decorator` to keep the context aware.

Reply:
Done. use mixed_precision.decorate